Sources/Rendering/OpenGL/RenderWindow/index.js

```diff
@@ -275,111 +278,143 @@ function vtkOpenGLRenderWindow(publicAPI, model) {
     return result;
   };
 
-  publicAPI.startVR = () => {
-    model.oldCanvasSize = model.size.slice();
-    if (model.vrDisplay.capabilities.canPresent) {
-      model.vrDisplay
-        .requestPresent([{ source: model.canvas }])
-        .then(() => {
-          if (
-            model.el &&
-            model.vrDisplay.capabilities.hasExternalDisplay &&
-            model.hideCanvasInVR
-          ) {
-            model.el.style.display = 'none';
-          }
-          if (model.queryVRSize) {
-            const leftEye = model.vrDisplay.getEyeParameters('left');
-            const rightEye = model.vrDisplay.getEyeParameters('right');
-            const width = Math.floor(
-              leftEye.renderWidth + rightEye.renderWidth
-            );
-            const height = Math.floor(
-              Math.max(leftEye.renderHeight, rightEye.renderHeight)
-            );
-            publicAPI.setSize(width, height);
-          } else {
-            publicAPI.setSize(model.vrResolution);
-          }
+  // Request an XR session on the user device with WebXR,
+  // typically in response to a user request such as a button press
+  publicAPI.startXR = () => {
+    if (navigator.xr === undefined) {
+      vtkErrorMacro('Could not start XR: not available in this browser!');
+      return;
+    }
 
-          const ren = model.renderable.getRenderers()[0];
-          ren.resetCamera();
-          model.vrFrameData = new VRFrameData();
-          model.renderable.getInteractor().switchToVRAnimation();
-
-          model.vrSceneFrame = model.vrDisplay.requestAnimationFrame(
-            publicAPI.vrRender
-          );
-          // If Broswer is chrome we need to request animation again to canvas update
-          if (IS_CHROME) {
-            model.vrSceneFrame = model.vrDisplay.requestAnimationFrame(
-              publicAPI.vrRender
-            );
-          }
```

Comment:
@martinken Based on testing it does not seem like we need to request animation twice for Chrome, is this consideration present elsewhere? Maybe a stopgap for a Chrome issue at the time of WebVR investigation?

Reply:
Removed.